File: 2015/2/2.exs
defmodule Day02 do
    def area(l, w, h) do
        2*l*w + 2*w*h + 2*h*l + l*w
    end
    def ribbon(l, w, h) do
        2*l + 2*w + l*w*h
    end

end

{:ok, raw } = File.read('input.txt')

input = raw |> String.split("\n", trim: true)

sum_array = Enum.map input, fn z ->
    [l, w, h] = z |> String.split("x", trim: true) |> Enum.map(&String.to_integer/1) |> Enum.sort
    Day02.area(l, w, h)
end
ribbon_array = Enum.map input, fn z ->
    [l, w, h] = z |> String.split("x", trim: true) |> Enum.map(&String.to_integer/1) |> Enum.sort
    Day02.ribbon(l, w, h)
end

IO.puts "Part1 : #{Enum.sum sum_array}"
IO.puts "Part2 : #{Enum.sum ribbon_array}"

File: 2021/2/2.py
import re
# f = open("input.txt")
f = open("test.txt")

# part1
h = 0
d = 0
for l in f:
  line = l.split()
  if re.search("^forward.*", line[0]):
    d = d + int(line[1])
  elif re.search("^down.*", line[0]):
    h = h + int(line[1])
  elif re.search("^up.*", line[0]):
    h = h - int(line[1])

print("part1:")
print(h*d)

# part2
f = open("input.txt")
# f = open("test.txt")
h = 0
d = 0
a = 0
for l in f:
  line = l.split()
  if re.search("^forward.*", line[0]):
    h = h + int(line[1])
    d = d + a*int(line[1])
  elif re.search("^down.*", line[0]):
    a = a + int(line[1])
  elif re.search("^up.*", line[0]):
    a = a - int(line[1])

print("part2:")
print(h*d)
